recipes-core/rcu-hostname/files/rcuhostname

```diff
@@ -0,0 +1,27 @@
+#!/bin/sh
+
+rcuserialpath=/usr/bin/rcu-serial
+tdxserialpath=/proc/device-tree/serial-number
+
+#set serial to Apalis SOM serial number
+if [ -f ${tdxserialpath} ]; then
+    serial=$(tr -d '\0' <${tdxserialpath})
+fi
+
+#if RCU serial exists, we prefer the RCU serial number
+if [ -f ${rcuserialpath} ]; then
+    serial=$(tr -d '\0' <${rcuserialpath})
+fi
+
+#if serial number is still empty
+if [ -z "$serial" -a "$serial" != " "  ]; then
+    serial="EMPTYSERIAL"
+fi
+
+hostname="NI-RCU-"${serial}
```

Comment:
I would name it as NI-SmartRack-xxx because customer may not know what is actually an "RCU"

Reply:
Seems like can just handled by **if , elseIf , else** (according to priority - RCU serial, SoM serial , EmptySerial):
Please refer to below :

```
#set serial to RCU serial number
if [ -f ${rcuserialpath} ]; then
    serial=$(tr -d '\0' <${rcuserialpath})

#set serial to Apalis SOM serial number
elif [ -f ${tdxserialpath} ]; then
     serial=$(tr -d '\0' <${tdxserialpath})

#set serial to "EMPTYSERIAL"
else
    serial="EMPTYSERIAL"
fi
```

Reply:
Referring to Toradex's template for this script, it seems like there's a chance our serial file will exist but somehow empty or corrupted. If we proceed with the if elif else construct, I'm afraid we might not be able to account for that case.